app/packages/dataset/src/Dataset.tsx

```diff
@@ -57,18 +58,35 @@ export interface DatasetProps {
   theme?: "dark" | "light";
   toggleHeaders?: () => void;
   canEditSavedViews?: boolean;
-  savedViewSlug?: string;
 }
 
-export const Dataset: React.FC<DatasetProps> = ({
+export const Dataset: React.FC<DatasetProps> = (props) => {
+  const [environment] = useState(fos.getEnvironment);
```

Comment:
Why `useState` here, if there's no setter for it? Perhaps a `useMemo()` might be useful?
`const environment = React.useMemo(() => fos.getEnvironment(), []);`

Reply:
updated! thanks

I need to look into `useMemo()`. I'm still very unfamiliar with react